tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,103 @@
+# Weight Clustering
+
+This document provides an overview on weight clustering to help you determine how it fits with your use case. To dive right into the code, see the [Clustering with Keras](clustering_with_keras.ipynb) tutorial and the [API docs](../../api_docs/python). For additional details on how to use the Keras API, a deep dive into weight clustering, and documentation on more advanced usage patterns, see the [API usage guide](train_clustered_models.md).
+
+## Overview
+
+Clustering, or weight sharing, is a technique for reducing the number of unique weight values in a model. It first groups the weights of each layer into *N* clusters, then shares the cluster's centroid value for all the weights belonging to the cluster.
+
+This technique brings improvements in terms of model compression. It reduces the number of bits used to represent the weights, thus saving memory bandwidth. This can be critical for deploying deep learning models on embedded systems with limited resources.
```

Comment:
Does it reduce the number of bits used to represent the weights?

Thought it'd still be float32.

> ... saving memory bandwidth
We can use a similar phrasing from the proposal RFC to first state what benefits apply to everyone. It should then be clarified that memory bandwidth saving requires framework support, similar to what's mentioned for pruning. https://www.tensorflow.org/model_optimization/guide/pruning